public/components/event_analytics/explorer/events_views/data_grid.tsx

```diff
@@ -3,148 +3,271 @@
+      return columns;
+    }
+    return [];
+  }, [storedSelectedColumns]);
+
+  // used for which columns are visible and their order
+  const dataGridColumnVisibility = useMemo(() => {
+    if (storedSelectedColumns.length > 0) {
+      const columns: string[] = [];
+      storedSelectedColumns.map(({ name }) => {
+        columns.push(name);
+      });
+      return {
+        visibleColumns: columns,
+        setVisibleColumns: (visibleColumns: string[]) => {
+          // TODO: implement with sidebar field order (dragability) changes
+        },
+      };
+    }
+    // default shown fields
+    return {
+      visibleColumns: [],
+      setVisibleColumns: () => {},
+    };
+  }, [storedSelectedColumns]);
+
+  // sets the very first column, which is the button used for the flyout of each row
+  const dataGridLeadingColumns = useMemo(() => {
+    return [
+      {
+        id: 'inspectCollapseColumn',
+        headerCellRender: () => null,
+        rowCellRender: ({ rowIndex }: { rowIndex: number }) => {
+          return (
+            <FlyoutButton
+              ref={null}
+              http={http}
+              key={null}
+              docId={'undefined'}
+              doc={rows[rowIndex % pageFields.current[1]]}
+              selectedCols={explorerFields.queriedFields}
+              timeStampField={timeStampField}
+              explorerFields={explorerFields}
+              pplService={pplService}
+              rawQuery={rawQuery}
+              onFlyoutOpen={onFlyoutOpen}
+            />
+          );
+        },
+        width: 40,
+      },
+    ];
+  }, [rows, http, explorerFields, pplService, rawQuery, onFlyoutOpen]);
+
+  // renders what is shown in each cell, i.e. the content of each row
+  const dataGridCellRender = useCallback(
+    ({ rowIndex, columnId }: { rowIndex: number; columnId: string }) => {
+      const trueIndex = rowIndex % pageFields.current[1];
+      if (trueIndex < rows.length) {
+        if (columnId === '_source') {
+          return (
+            <EuiDescriptionList type="inline" compressed>
+              {Object.keys(rows[trueIndex]).map((key) => (
+                <Fragment key={key}>
+                  <EuiDescriptionListTitle className="osdDescriptionListFieldTitle">
+                    {key}
+                  </EuiDescriptionListTitle>
+                  <EuiDescriptionListDescription
+                    dangerouslySetInnerHTML={{ __html: dompurify.sanitize(rows[trueIndex][key]) }}
+                  />
+                </Fragment>
+              ))}
+            </EuiDescriptionList>
+          );
+        }
+        if (columnId === 'timestamp') {
+          return `${moment(rows[trueIndex][columnId]).format(DATE_DISPLAY_FORMAT)}`;
+        }
+        return `${rows[trueIndex][columnId]}`;
+      }
+      return null;
+    },
+    [rows, pageFields, explorerFields]
+  );
+
+  // ** Pagination config
+  const [pagination, setPagination] = useState({ pageIndex: 0, pageSize: 25 });
```

Comment:
Is this default value same as discover?

Reply:
Yes

Reply:
Seems like page size default for me is 100 - can we double check this/pull this value if it is set globall?

Reply:
You're right - changed the default page size to 100